### ISSUE TITLE
New roc_transceiver API

### ISSUE THREAD

**Issue by gavv** [roc-project/roc#259]
Currently we have roc_sender and roc_receiver API which has the following flow:

* roc_receiver binds to ports
* roc_sender connects to that ports
* the user writes the stream to roc_sender
* the user reads the stream mixed from all connected roc_senders from roc_receiver

Now we're implementing RTSP and this API causes two limitations:

* the stream is always unidirectional
* the sender is always a client (the one who connects) and the receiver is always a server (the one who binds)

To avoid this limitation, we're replacing roc_sender and roc_receiver with a single roc_transceiver object with the following flow:

* roc_transceiver has both bind and connect methods
* the user creates two roc_transceivers
* the user binds one of them to an endpoint (so it acts as a server)
* the user connects another one to the endpoint of the first one (so it acts as a client)
* the user can write and read from both of them
* if roc_transceiver acts as a client, it writes and reads the stream from the server to which it is connected
* if roc_transceiver acts as a server, the stream written to it is duplicated to all connected clients, and the stream read from it is mixed from all connected clients

So, in other words, roc_transceiver acts both as server and client, and both as sender and receiver, and also can be bidirectional.

The new API will be very similar to current roc_sender and roc_receiver so the migration should be straightforward (but still the compatibility will be broken).

**Comment by gavv** [comment]
Should be done after #258.
